ddht/v5_1/alexandria/seeker.py

```diff
@@ -0,0 +1,137 @@
+from typing import List, Set
+
+from async_service import Service
+from eth_enr import ENRAPI
+from eth_typing import NodeID
+from eth_utils import get_extended_debug_logger
+import trio
+
+from ddht._utils import reduce_enrs
+from ddht.kademlia import compute_distance
+from ddht.v5_1.alexandria.abc import AlexandriaNetworkAPI
+from ddht.v5_1.alexandria.content import content_key_to_content_id
+from ddht.v5_1.alexandria.typing import ContentID, ContentKey
+
+
+class Seeker(Service):
+    content_key: ContentKey
+    content_id: ContentID
+
+    _content: bytes
+    _content_ready: trio.Event
+
+    def __init__(
+        self,
+        network: AlexandriaNetworkAPI,
+        content_key: ContentKey,
+        concurrency: int = 3,
+    ) -> None:
+        self.logger = get_extended_debug_logger("ddht.Seeker")
+
+        self.concurrency = concurrency
+
+        self.content_key = content_key
+        self.content_id = content_key_to_content_id(content_key)
+
+        self._network = network
+
+        self._content_send, self.content_receive = trio.open_memory_channel[bytes](0)
+
+    async def run(self) -> None:
+        async with self.content_receive, self._content_send:
+            raw_enr_send, raw_enr_receive = trio.open_memory_channel[ENRAPI](32)
+            sorted_enr_send, sorted_enr_receive = trio.open_memory_channel[ENRAPI](0)
+
+            self.manager.run_daemon_task(self._explore_for_enrs, raw_enr_send)
+            self.manager.run_daemon_task(
+                self._collate, sorted_enr_send, raw_enr_receive
+            )
+
+            for _ in range(self.concurrency):
+                self.manager.run_daemon_task(
+                    self._worker, raw_enr_send, sorted_enr_receive,
```

Comment:
Yeah, I think I might have breezed by this method if it had been something like:
```suggestion
                    self._seek, recent_enr_send, closest_enr_receive,
```